lua/telescope/finders.lua

```diff
@@ -14,6 +17,58 @@ local _callable_obj = function()
   return obj
 end
 
+local static_find
+do
+  local exe = Executor.new {}
+  static_find = function(results, process_result, process_complete)
+    if #exe.tasks ~= 0 then
+      exe:close()
+    end
+    exe:add(co.create(function(results)
+      for _, v in ipairs(results) do
+        process_result(v)
```

Comment:
One thing we can do is also check the result of `process_result` or similar to determine if we should stop the execution.